gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/model/Subscription.java

```diff
@@ -74,8 +76,16 @@ public enum AuditEvent implements Audit.ApiAuditEvent {
      */
     private String reason;
 
+    /**
+     * The status of the current subscription
+     */
     private Status status;
 
+    /**
+     * The status chosen by the consumer, used to know who PAUSED a subscription
+     */
+    private ConsumerStatus consumerStatus;
```

Comment:
we think it should bring extra complexity and, with only one status, how do you manage the fact consumer put the subscription in pause and the publisher do it because of a maintenance or whatever.
If consumer resume, the subscription should stay inactive because it's still paused from a publisher point of view.
Having two statuses allows to allow overlap of pause period

Reply:
With one field I would think on ACTIVE / CONSUMER_STOPPED / PUBLISHER_STOPPED / FAILED status.

Reply:
it will haevily impact the existing part, so maybe we can discuss it in a meet